includes/AMP/Sanitization.php

```diff
@@ -0,0 +1,459 @@
+		 * for example, when using Google Fonts."
+		 *
+		 * Note that \AMP_Style_Sanitizer::process_link_element() will ensure preconnect links for Google Fonts are present.
+		 */
+		$link_relations = [ Attribute::REL_PRECONNECT, Attribute::REL_DNS_PREFETCH, Attribute::REL_PRELOAD, Attribute::REL_PRERENDER, Attribute::REL_PREFETCH ];
+		foreach ( $link_relations as $rel ) {
+			if ( ! isset( $links[ $rel ] ) ) {
+				continue;
+			}
+
+			/**
+			 * Link node.
+			 *
+			 * @var DOMElement $link
+			 */
+			foreach ( $links[ $rel ] as $link ) {
+				if ( $link->parentNode ) {
+					$link->parentNode->removeChild( $link ); // So we can move it.
+				}
+				if ( $previous_node && $previous_node->nextSibling ) {
+					$document->head->insertBefore( $link, $previous_node->nextSibling );
+					$previous_node = $link;
+				}
+			}
+		}
+
+		// "5. Load the AMP runtime."
+		if ( isset( $amp_scripts[ Amp::RUNTIME ] ) ) {
+			$ordered_scripts[ Amp::RUNTIME ] = $amp_scripts[ Amp::RUNTIME ];
+			unset( $amp_scripts[ Amp::RUNTIME ] );
+		} else {
+			$script = $document->createElement( Tag::SCRIPT );
+			$script->setAttribute( Attribute::ASYNC, '' );
+			$script->setAttribute( Attribute::SRC, $runtime_src );
+			$ordered_scripts[ Amp::RUNTIME ] = $script;
+		}
+
+		/*
+		 * "6. Specify the <script> tags for render-delaying extensions (e.g., amp-experiment amp-dynamic-css-classes and amp-story"
+		 *
+		 * {@link https://amp.dev/documentation/guides-and-tutorials/optimize-and-measure/optimize_amp/ AMP Hosting Guide}
+		 */
+		foreach ( Amp::RENDER_DELAYING_EXTENSIONS as $extension ) {
+			if ( isset( $amp_scripts[ $extension ] ) ) {
+				$ordered_scripts[ $extension ] = $amp_scripts[ $extension ];
+				unset( $amp_scripts[ $extension ] );
+			}
+		}
+
+		/*
+		 * "7. Specify the <script> tags for remaining extensions (e.g., amp-bind ...). These extensions are not render-delaying
+		 * and therefore should not be preloaded as they might take away important bandwidth for the initial render."
+		 */
+		ksort( $amp_scripts );
+		$ordered_scripts = array_merge( $ordered_scripts, $amp_scripts );
+		/**
+		 * Script element.
+		 *
+		 * @var DOMElement $ordered_script
+		 */
+		foreach ( $ordered_scripts as $ordered_script ) {
+			if ( $previous_node && $previous_node->nextSibling ) {
+				$document->head->insertBefore( $ordered_script, $previous_node->nextSibling );
+				$previous_node = $ordered_script;
+			}
+		}
+	}
+
+	/**
+	 * Returns AMP extension URLs, keyed by extension name.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @return array List of extensions and their URLs.
+	 */
+	protected function get_extension_sources() {
+		$specs = [];
+		// Register all AMP components as defined in the spec.
+		foreach ( AMP_Allowed_Tags_Generated::get_extension_specs() as $extension_name => $extension_spec ) {
+			$src = sprintf(
+				'https://cdn.ampproject.org/v0/%s-%s.js',
+				$extension_name,
+				end( $extension_spec['version'] )
```

Comment:
Beware that the `end()` is not always correct. At the moment, the `1.0` version of `amp-experiment` is the largest version, but is still experimental. The `0.1` version needs to be used instead: https://github.com/ampproject/amp-wp/blob/8856284d90fc8558c30acc029becd352ae26e4e1/includes/amp-helper-functions.php#L1051-L1073

I suppose `amp-experiment` is not relevant for Stories, however.

Reply:
Can add this part 👍